app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -505,6 +505,17 @@ class BrowserTabViewModel(
         viewModelScope.launch { removeCurrentTabFromRepository() }
     }
 
+    override fun closeAndSelectSourceTab() {
+        viewModelScope.launch { removeAndSelectTabsFromRepository() }
+    }
+
+    private suspend fun removeAndSelectTabsFromRepository() {
+        val currentTab = tabRepository.liveSelectedTab.value
```

Comment:
Sharing some thoughts with you here:

I see that we obtain the `currentTab` directly from `tabRepository.liveSelectedTab`, we do that here and in `BrowserViewModel`. I wonder if it could be better to move that logic inside the Repository, and directly work with `TabsDao.selectedTab()` from there. That way we will reduce the code in the `ViewModels` and put that logic close to the source of data. What do you think? For example `deleteAndSelectSource` could be renamed to `deleteCurrentTabAndSelectSource()`.

Reply:
Makes sense and will definitely reduce some unnecessary code.